server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -452,3 +474,24 @@ func tempCopy(r io.Reader) (path string, err error) {
 	}
 	return f.Name(), nil
 }
+
+type dialerControl = func(network, address string, conn syscall.RawConn) error
+
+func blockingDialerControl(allowed []*net.IPNet) dialerControl {
+	return func(network, address string, conn syscall.RawConn) error {
+		host, _, err := net.SplitHostPort(address)
+		if err != nil {
+			return err
+		}
+		ip := net.ParseIP(host)
+		for _, ipNet := range allowed {
+			if ipNet.Contains(ip) {
+				return nil
+			}
+		}
+		if ip.IsLoopback() || ip.IsPrivate() || ip.IsLinkLocalUnicast() || ip.IsLinkLocalMulticast() {
+			return errors.New("IP address not allowed")
```

Comment:
We could debug log the IP address here. Returning it in the error may not be a good idea as that could be used to perform "privileged" DNS resolution.

Reply:
added an info-level log since these logs will probably be sparse and it seems useful to have these available without debug logging turned on.